src/cmd/cmd.zig

```diff
@@ -618,23 +617,53 @@ fn getMyDataFromIpEcho(
 }
 
 fn getEntrypoints(logger: Logger) !std.ArrayList(SocketAddr) {
+    const EntrypointSet = std.ArrayHashMap(void, void, struct {
+        // zig fmt: off
+        pub fn hash(_: @This(), _: void) u32 { unreachable; }
+        pub fn eql(_: @This(), _: void, _: void, _: usize) bool { unreachable; }
+        // zig fmt: on
+    }, true);
+    const EntrypointCtx = struct {
+        entrypoints: []const SocketAddr,
+        pub fn hash(_: @This(), entrypoint: SocketAddr) u32 {
+            const array, const len = entrypoint.toString();
+            return std.array_hash_map.hashString(array[0..len]);
+        }
+        pub fn eql(ctx: @This(), a: SocketAddr, _: void, b_index: usize) bool {
+            return a.eql(&ctx.entrypoints[b_index]);
+        }
+    };
+
     var entrypoints = std.ArrayList(SocketAddr).init(gpa_allocator);
+    errdefer entrypoints.deinit();
+
+    var entrypoint_set = EntrypointSet.init(gpa_allocator);
+    defer entrypoint_set.deinit();
+
+    try entrypoint_set.ensureTotalCapacity(config.current.gossip.entrypoints.len);
+    try entrypoints.ensureTotalCapacityPrecise(config.current.gossip.entrypoints.len);
```

Comment:
why do you need both of these?

Reply:
If you mean the variables, `entrypoint_set` is a set of entrypoints that have already been set. I ensure capacity for both so that neither have to reallocate during the loop.

Reply:
oh sorry didnt notice the different variable names